server/android/src/android.rs

```diff
@@ -2,167 +2,89 @@
-    struct ServerBucket {
-        // next handle ready to be used
-        next_handle: i64,
-        map: HashMap<i64, RunningServerHandle>,
-    }
-
-    impl ServerBucket {
-        /// Returns an integer handle to the running server
-        fn insert(&mut self, server: RunningServerHandle) -> i64 {
-            let handle = self.next_handle;
-            self.next_handle = handle + 1;
-            self.map.insert(handle, server);
-            handle
-        }
-    }
-
-    static SERVER_BUCKET: Lazy<Mutex<ServerBucket>> = Lazy::new(|| {
-        Mutex::new(ServerBucket {
-            next_handle: 1,
-            map: HashMap::new(),
-        })
-    });
-
-    fn generate_certs(cert_dir: &PathBuf) {
-        let subject_alt_names = vec!["localhost".to_string()];
-        let cert = generate_simple_self_signed(subject_alt_names).unwrap();
-
-        std::fs::create_dir_all(cert_dir).unwrap();
-        let mut file = std::fs::File::create(cert_dir.join(PRIVATE_CERT_FILE)).unwrap();
-        file.write_all(cert.serialize_private_key_pem().as_bytes())
-            .unwrap();
-
-        let mut file = std::fs::File::create(cert_dir.join(PUBLIC_CERT_FILE)).unwrap();
-        file.write_all(cert.serialize_pem().unwrap().as_bytes())
-            .unwrap();
-    }
-
-    /// Bindings test method. TODO: remove when not needed anymore
-    #[no_mangle]
-    pub unsafe extern "C" fn Java_org_openmsupply_client_RemoteServer_rustHelloWorld(
-        env: JNIEnv,
-        _: JClass,
-        name: JString,
-    ) -> jstring {
-        let name: String = env.get_string(name).unwrap().into();
-        let response = format!("Hello {}!", name);
-        env.new_string(response).unwrap().into_inner()
-    }
+    static SERVER_BUCKET: Mutex<Option<ServerBucket>> = Mutex::new(None);
 
     #[no_mangle]
-    pub unsafe extern "C" fn Java_org_openmsupply_client_RemoteServer_startServer(
+    pub unsafe extern "C" fn Java_org_openmsupply_mobile_RemoteServer_startServer(
         env: JNIEnv,
         _: JClass,
         port: jchar,
         files_dir: JString,
         cache_dir: JString,
         android_id: JString,
-    ) -> jlong {
-        android_logger::init_once(Config::default().with_min_level(Level::Trace));
+    ) {
+        android_logger::init_once(Config::default().with_min_level(Level::Info));
+        log_panics::init();
 
         let (off_switch, off_switch_receiver) = mpsc::channel(1);
         let files_dir: String = env.get_string(files_dir).unwrap().into();
         let files_dir = PathBuf::from(&files_dir);
         let android_id: String = env.get_string(android_id).unwrap().into();
         let db_path = files_dir.join("omsupply-database");
-
         let cache_dir: String = env.get_string(cache_dir).unwrap().into();
 
-        let cert_path = files_dir.join("certs");
-        if !cert_path.join(PRIVATE_CERT_FILE).exists() {
-            generate_certs(&cert_path);
```

Comment:
We already generate this in server if danger_allow_http: false, once again I think this wasn't the case when was originally written

Reply:
thanks! this was on my todo list when these changes where made...